File: go/internal/usecase/getlatest/execute.go
package getlatest

import (
	"context"

	"github.com/dtorres47/fuel-downloader/go/internal/domain"
	"github.com/dtorres47/fuel-downloader/go/internal/infra/csvexp"
	"github.com/dtorres47/fuel-downloader/go/internal/infra/eia"
	"github.com/dtorres47/fuel-downloader/go/internal/infra/postgres"
)

type Deps struct {
	EIA  *eia.Client
	Repo *postgres.Repo
}

func Execute(ctx context.Context, d Deps, area, outPath string) (domain.FuelRate, error) {
	fr, err := d.EIA.FetchLatestDiesel(ctx, area)
	if err != nil {
		return domain.FuelRate{}, err
	}
	if err := d.Repo.Upsert(ctx, fr); err != nil {
		return domain.FuelRate{}, err
	}
	if err := csvexp.Write(outPath, fr); err != nil {
		return domain.FuelRate{}, err
	}
	return fr, nil
}

File: csharp/src/FuelDownloader.UseCase/GetLatest/Execute.cs
using FuelDownloader.Domain;
using FuelDownloader.Infra.Eia;
using FuelDownloader.Infra.Postgres;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FuelDownloader.UseCase.GetLatest
{
    public class Executor
    {
        private readonly Client _client;
        private readonly Repo _repo;

        public Executor(Client client, Repo repo)
        {
            _client = client;
            _repo = repo;
        }

        // Placeholder: fetch → upsert → export
        public async Task<FuelRate?> ExecuteAsync(string outputPath, string area = "NUS")
        {
            // TODO: fetch from EIA
            // TODO: upsert into Postgres
            // TODO: write CSV
            await Task.CompletedTask;
            return null;
        }
    }
